src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step/pattern_review_step.js

```diff
@@ -24,35 +26,51 @@ modules.get('apps/settings')
       controllerAs: 'reviewStep',
       bindToController: true,
       controller: function ($scope, Private) {
+        this.errors = [];
         if (_.isUndefined(this.indexPattern)) {
           this.indexPattern = {};
         }
 
         const knownFieldTypes = {};
         this.dateFields = [];
-        this.pipeline.model.processors.forEach((processor) => {
-          if (processor.typeId === 'geoip') {
-            const field = processor.targetField || 'geoip';
-            knownFieldTypes[field] = 'geo_point';
+        if (this.pipeline) {
+          this.pipeline.model.processors.forEach((processor) => {
+            if (processor.typeId === 'date') {
+              const field = processor.targetField || '@timestamp';
+              knownFieldTypes[field] = 'date';
+              this.dateFields.push(field);
+            }
+          });
+        }
+
+        const fields = {};
```

Comment:
How about renaming `fields` to `sampleFields` or `sampleDocFields`? I think this is a more informative name since `fields` contains values from `sampleDoc`. It also better separates it from `indexPatternFields` which is named specifically to tell the reader that it contains the field data from the index pattern.